packages/compass-settings/src/components/modal.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import ipc from 'hadron-ipc';
+
+import {
+  Modal,
+  ModalTitle,
+  ModalFooter,
+  H3,
+  css,
+  spacing,
+  Button,
+} from '@mongodb-js/compass-components';
+
+import { fetchSettings } from '../stores/settings';
+
+import PrivacySettings from './settings/privacy';
+import Sidebar from './sidebar';
+
+type Settings = {
+  name: string;
+  component: React.ComponentType;
+};
+
+type SettingsModalProps = {
+  onInit: () => void;
+};
+
+const contentStyles = css({
+  display: 'flex',
+  minHeight: '400px',
+});
+
+const sideNavStyles = css({
+  width: '20%',
+});
+
+const settingsStyles = css({
+  width: '80%',
+  padding: spacing[2],
+});
+
+const settings: Settings[] = [{ name: 'Privacy', component: PrivacySettings }];
+
+export const SettingsModal: React.FunctionComponent<SettingsModalProps> = ({
+  onInit,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedSetting, setSelectedSettings] = useState(settings[0].name);
+
+  const SettingComponent =
+    settings.find((x) => x.name === selectedSetting)?.component ?? null;
+
+  useEffect(() => {
+    if (isOpen) {
+      onInit();
+    }
+  }, [onInit, isOpen]);
+
+  useEffect(() => {
+    (ipc as any).on('window:show-network-optin', () => {
+      setIsOpen(!isOpen);
+    });
+  }, [setIsOpen, isOpen]);
```

Comment:
This will recreate the subscription every time `isOpen` is flipped. If you need this to happen, you need to either use a return value to unsubscribe, or use a callback inside `setState` instead of directly refererring to value, which is also [a recommeded approach](https://reactjs.org/docs/hooks-reference.html#functional-updates) when new state depends on the previous one

I would also maybe consider not doing this, FWIW this is really not a common pattern in the applications to use menu as a toggle, unless the label in the menu explicitly says "Toggle <smth>"

Reply:
I removed the toggle completely.